src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -311,6 +311,11 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
     private fun checkImpl(holder: AnnotationHolder, impl: RsImplItem) {
         val trait = impl.traitRef?.resolveToTrait ?: return
         val traitName = trait.name ?: return
+        if (impl.unsafe != null && trait.unsafe == null) {
+            holder.createErrorAnnotation(impl.traitRef!!, "Safe traits should not have unsafe implementations [E0199]")
```

Comment:
We usually try to stick to the exact (or at least as close as we can) error messages that the compiler emits. In these cases they are "Implementing the trait \`Foo\` is not unsafe [E0199]" and "The trait \`Foo\` requires an \`unsafe impl\` declaration [E0200]". Could you, please, change them accordingly?